src/js/theme/template-parts.js

```diff
@@ -0,0 +1,327 @@
+import { Window as window } from '../utils/server-safe-globals.js';
+
+/* Adapted from https://github.com/dy/template-parts - ISC - Dmitry Iv. */
+
+// Template Instance API
+// https://github.com/WICG/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md
+
+const ELEMENT = 1,
+  STRING = 0,
+  PART = 1;
```

Comment:
nit: having them be separate consts would make changing them easier, should we need to (less diff churn)
```suggestion
const ELEMENT = 1;
const STRING = 0;
const PART = 1;
```